From 1225b41f700799c471f6aa66021da0c72fcd47a9 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Tue, 23 Aug 2016 16:16:21 -0700
Subject: [PATCH] joyent/eng#10 Explain https URI preferences for submodules
 and npm deps joyent/eng#9 Update boilerplate deps, tap => tape

---
 .gitmodules      |  6 +++---
 docs/index.md    | 29 +++++++++++++++++++++++++++++
 package.json     | 10 +++++-----
 test/api.test.js |  2 +-
 4 files changed, 38 insertions(+), 9 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index dbadc87..1d72cc4 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,9 +1,9 @@
 [submodule "deps/restdown"]
 	path = deps/restdown
-	url = git://github.com/trentm/restdown.git
+	url = https://github.com/trentm/restdown.git
 [submodule "deps/javascriptlint"]
 	path = deps/javascriptlint
-	url = git://github.com/davepacheco/javascriptlint.git
+	url = https://github.com/davepacheco/javascriptlint.git
 [submodule "deps/jsstyle"]
 	path = deps/jsstyle
-	url = git://github.com/davepacheco/jsstyle.git
+	url = https://github.com/davepacheco/jsstyle.git
diff --git a/docs/index.md b/docs/index.md
index e96e4aa..107171e 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -182,6 +182,35 @@ differ (e.g. not have a test suite):
 There are several modular Makefiles you can use to implement most of this. See
 "Writing Makefiles" (below) for details.
 
+
+### package.json and git submodules
+
+Repositories containing node.js code should have a `package.json` file at the
+root of the repository. Normally most dependencies should be taken care of
+through `npm` and this `package.json` file, by adding entries to the
+`dependencies` or `devDependencies` arrays (see examples in this repository, or
+documentation on the `npm` website).
+
+Dependencies via `npm` can either take the form of an `npm` package name (in
+which case it must be published to the public `npm` package servers), or a `git`
+URL. We have a weak preference towards public `npm` packages, stronger for
+shared library code that could be used outside Triton proper.
+
+If using a `git` URL in an `npm` dependency, you must use a `git+https://` URL
+to specify it (not `git://` or `git+ssh://`). Plain `git://` operations are not
+authenticated in any way and can be hijacked by malicious WiFi or other network
+man-in-the-middle attacks (e.g. at airports and coffee shops - yes, this
+actually happens). The use of `git+ssh://` URLs is discouraged because it
+prevents users from being able to clone and build the package on a machine that
+does not have their GitHub private key on it. In particular, some Joyent CI bots
+(like the Gerrit `make check` bot) will complain if you use these URLs.
+
+For certain dependencies, it is standard practice across the Joyent
+repositories to use `git` submodules instead. This applies in particular to
+`javascriptlint`, `jsstyle`, `restdown` and `sdc-scripts`. Similar to `npm`
+`git` dependencies, these must use `https://` URLs only.
+
+
 ## Coding Style
 
 Every repository **must** have a consistent coding style that is enforced by
diff --git a/package.json b/package.json
index ff23943..231da43 100644
--- a/package.json
+++ b/package.json
@@ -6,16 +6,16 @@
   "private": true,
   "dependencies": {
     "restify": "2.2.1",
-    "node-uuid": "1.4.0",
-    "filed": "0.0.5",
-    "bunyan": "0.18.2"
+    "node-uuid": "^1.4.0",
+    "bunyan": "^1.8.0",
+    "git-example": "git+https://github.com/joyent/node-git-example#a3216ae18f384717b7be7148cc151ef743de45d5"
   },
   "devDependencies": {
-    "tap": "0.4.0"
+    "tape": "^4.6.0"
   },
   "scripts": {
     "start": "node ./server.js",
-    "test": "./node_modules/.bin/tap test/*.test.js"
+    "test": "./node_modules/.bin/tape test/*.test.js"
   },
   "license": "MPL-2.0"
 }
diff --git a/test/api.test.js b/test/api.test.js
index 8ac6a1b..c1228a0 100644
--- a/test/api.test.js
+++ b/test/api.test.js
@@ -10,7 +10,7 @@
 
 /* Test the Boilerplate API endpoints */
 
-var test = require('tap').test;
+var test = require('tape').test;
 
 
 test('douglas adams', function (t) {
-- 
2.21.0

